src/customFunctions/customFetch.ts

```diff
@@ -0,0 +1,43 @@
+import { bearerTokenAuthorization } from "./../configs/bearerTokenAuthorization";
+
+type CustomFetch<TResponseJson> = {
+  fullResponse: Response;
+  json: TResponseJson;
+  text: string;
+};
+
+export const customFetch = async <TResponseJson>(
+  input: RequestInfo,
+  init: RequestInit = {}
+): Promise<CustomFetch<TResponseJson>> => {
+  init.headers = Object.assign(bearerTokenAuthorization.headers, init.headers);
+  const response = await fetch(input, init);
+
+  const isJson = response.headers
+    .get("content-type")
+    ?.includes("application/json");
+  const data = isJson ? await response.clone().json() : null;
+
+  if (!response.ok) {
+    const error = (data && data.message) || response.status;
+    throw new HttpResponseError("Some problem with response: ", error);
+  }
+
+  return {
+    /*  NOTE: clone() method creates a copy. 
+        Both methods of creating a copy will fail if the body of the original request 
+        or response has already been read, but reading the body of a cloned response
+        or request will not cause it to be marked as read in the original.
+        https://developer.mozilla.org/en-US/docs/Web/API/Fetch_API/Using_Fetch
+     */
+    fullResponse: response.clone(),
+    json: data,
+    text: await response.clone().text(),
```

Comment:
isJson je famozní počin, nečekal jsem, že to tam zapracuješ, mega dobrý.

ale je dost zbytečný parsovat všechny věci, když je pak user nepoužije, to je dost neoptimalizovaný, já bych prostě vrátil data, který když content-type je json, tak budou rozparsovaný jako json, když to nebude json, tak další třeba .text(), další možnost jak může člověk parsovat response je třeba blob, ale to je v kontextu SPA zbytečný imho řešit, takže bych to nechal jako buď text, nebo json v závislosti na content-typu, to je podle mě fair... nebo je varianta když conent type je json rozparsovat to jako json a když není, tak to nechat uživatele té knihovny, ať si to sám rozparuje, aby měl low level přístup i na blob, protože je otázka, co se stane, pokuď příjde blob a ty zavoláš `.text()`, jeslti to náhodou nevyhodí runtime error a celý to nafailne